### PR TITLE
Support trailing underscores in usernames

### PR DESCRIPTION
**Pull request by babolivier**
Since this is my first PR on this project, please let me know if this isn't the right way to do it, or if I've forgotten anything.

Fixes https://github.com/matrix-org/matrix-appservice-irc/issues/1359

### CODE REVIEW

**Comment by jaller94** [comment]
Isn't it that underscores shouldn't get sanitised by `sanitiseUsername` if they don't fit the delimiter pattern of the bridge?

File: src/bridge/AdminRoomHandler.ts [hunk 1]
```diff
@@ -480,7 +482,7 @@ export class AdminRoomHandler {
                     `Username is longer than the maximum permitted by the bridge (${SANE_USERNAME_LENGTH}).`
                 );
             }
-            else if (IdentGenerator.sanitiseUsername(username) !== username) {
+            else if (IdentGenerator.sanitiseUsername(username) !== usernameWithoutUnderscores) {
```
Comment (Half-Shot) [inline src/bridge/AdminRoomHandler.ts:485]: I think the best approach might be to make santitiseUsername more lax to be honest. The linked issue seems to mention that more characters are supported than I expected, so perhaps just make that accept more?

**Review by Half-Shot — COMMENTED** [review]
Thanks for the PR! I think the approach would fix this specific issue, but I'd like to improve our sanitise function :)

**Comment by babolivier** [comment]
Superceded by https://github.com/matrix-org/matrix-appservice-irc/pull/1538

**Comment by Half-Shot** [comment]
And now superseded by https://github.com/matrix-org/matrix-appservice-irc/pull/1719

**Pull request closed without merging**